src/usr/local/www/interfaces_gre_edit.php

```diff
@@ -175,32 +227,66 @@ function build_parent_list() {
 	$pconfig['remote-addr']
 ))->setHelp('Peer address where encapsulated GRE packets will be sent.');
 
-$section->addInput(new Form_IpAddress(
+$group = new Form_Group(gettext('IPv4'));
+
+$group->add(new Form_IpAddress(
 	'tunnel-local-addr',
 	'*GRE tunnel local address',
 	$pconfig['tunnel-local-addr']
 ))->setHelp('Local GRE tunnel endpoint.');
 
-$section->addInput(new Form_IpAddress(
+$group->add(new Form_IpAddress(
 	'tunnel-remote-addr',
 	'*GRE tunnel remote address',
 	$pconfig['tunnel-remote-addr']
 ))->setHelp('Remote GRE address endpoint.');
 
-$section->addInput(new Form_Select(
+$group->add(new Form_Select(
 	'tunnel-remote-net',
 	'*GRE tunnel subnet',
 	$pconfig['tunnel-remote-net'],
-	array_combine(range(128, 1, -1), range(128, 1, -1))
+	array_combine(range(32, 1, -1), range(32, 1, -1))
 ))->setHelp('The subnet is used for determining the network that is tunnelled.');
 
-$section->addInput(new Form_Checkbox(
+$group->add(new Form_Checkbox(
 	'link1',
 	'Add Static Route',
 	'Add an explicit static route for the remote inner tunnel address/subnet via the local tunnel address',
 	$pconfig['link1']
 ));
 
+$section->add($group);
+
+$group = new Form_Group(gettext('IPv6'));
+
+$group->add(new Form_IpAddress(
+	'tunnel-local-addr6',
+	'*GRE tunnel local v6 address',
+	$pconfig['tunnel-local-addr6']
+))->setHelp('Local v6 GRE tunnel endpoint.');
+
+$group->add(new Form_IpAddress(
+	'tunnel-remote-addr6',
+	'*GRE tunnel remote v6 address',
+	$pconfig['tunnel-remote-addr6']
+))->setHelp('Remote v6 GRE address endpoint.');
+
+$group->add(new Form_Select(
+	'tunnel-remote-net6',
+	'*GRE tunnel v6 subnet',
+	$pconfig['tunnel-remote-net6'],
+	array_combine(range(128, 1, -1), range(128, 1, -1))
+))->setHelp('The subnet is used for determining the v6 network that is tunnelled.');
+
+$group->add(new Form_Checkbox(
+	'link3',
+	'Add Static Route',
+	'Add an explicit static route for the remote inner tunnel address/subnet via the local tunnel address',
+	$pconfig['link3']
+));
```

Comment:
link0/link1/link2 have special meanings to `ifconfig` and GRE. We can't just make up a new value in this case.

The `link1` property is for the interface as a whole -- see https://github.com/pfsense/pfsense/blob/master/src/etc/inc/interfaces.inc#L1311 -- so you can just take the value of `link1` rather than making a new option.

Reply:
Ah, okay. This then leads me to the question why the link1 property is set when the user "only" requests a static route to be added? Should I use two distinct fields for each static route and set link1 when either of them is requested or should I really just put one option there for the static route and add a v4 as well as a v6 route when it's set?

Reply:
I am now only using link1 and set a static route for any active network.

Reply:
> This then leads me to the question why the link1 property is set when the user "only" requests a static route to be added?

Because the `link1` flag has a history in which it altered the internal route search method used for locating the remote endpoint. So we add that route manually to account for how GRE behaved in the past. That behavior has changed over time (See b5baa9e7a17266e794514747e7046eba6acb1db7 for example) but we kept the user-facing option the same to preserve what users expect over time.

Using just the one option is fine as the behavior is defined for the interface as a whole and shouldn't be a per-family option.

Reply:
Ah! I see, thank you for explaining it. :-)